sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala

```diff
@@ -133,8 +134,21 @@ case class StaticInvoke(
   override def nullable: Boolean = needNullCheck || returnNullable
   override def children: Seq[Expression] = arguments
 
-  override def eval(input: InternalRow): Any =
-    throw new UnsupportedOperationException("Only code-generated evaluation is supported.")
+  override def eval(input: InternalRow): Any = {
+    if (staticObject == null) {
+      throw new RuntimeException("The static class cannot be null.")
+    }
+
+    val parmTypes = arguments.map(e =>
+      CallMethodViaReflection.typeMapping.getOrElse(e.dataType,
+        Seq(e.dataType.asInstanceOf[ObjectType].cls))(0))
+    val parms = arguments.map(e => e.eval(input).asInstanceOf[Object])
+    val method = staticObject.getDeclaredMethod(functionName, parmTypes : _*)
```

Comment:
Looks like we can also move `getDeclaredMethod` outside `eval`?

Reply:
Unfortunately, this movement caused failure of reflection.